src/realm/query_engine.cpp

```diff
@@ -51,6 +51,9 @@ size_t ParentNode::find_first(size_t start, size_t end)
 
 void ParentNode::aggregate_local_prepare(Action TAction, DataType col_id, bool nullable)
 {
+    // Verify that the cached column accessor is still valid
+    verify_column(); // throws
```

Comment:
Good test cases are much more useful than review. The issue of a column having been removed is unlikely to be at the front of people's minds when they're modifying or reviewing query code in the future, making it easy to miss problems that are introduced. Exhaustive test coverage is a much more reliable way to prevent future problems.